

<!DOCTYPE html>


<html lang="en" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>Pr√©parer les donn√©es üóÉÔ∏è &#8212; Unix tutorial</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="../_static/styles/bootstrap.css?digest=e353d410970836974a52" rel="stylesheet" />
<link href="../_static/styles/pydata-sphinx-theme.css?digest=e353d410970836974a52" rel="stylesheet" />

  
  <link href="../_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=e353d410970836974a52" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=e353d410970836974a52" />
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52" />

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'tutorial_2/2_donnees';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
  

<a class="navbar-brand logo" href="../intro.html">
  
  
  
  
    
    
      
    
    
    <img src="../_static/logo.png" class="logo__image only-light" alt="Logo image"/>
    <script>document.write(`<img src="../_static/logo.png" class="logo__image only-dark" alt="Logo image"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../intro.html">
                    Bienvenue !
                </a>
            </li>
        </ul>
        <ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../connection.html">Connexion au JupyterHub de l‚ÄôIFB</a></li>
<li class="toctree-l1"><a class="reference internal" href="../tutorial_1/tutorial.html">Tuto 1 : un  aper√ßu du shell Unix</a></li>
<li class="toctree-l1"><a class="reference internal" href="0_intro.html">Tuto 2 : analyse des donn√©es RNA-seq de <em>S. cerevisiae</em> avec Unix</a></li>
<li class="toctree-l1"><a class="reference internal" href="../more.html">Ressources compl√©mentaires</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/pierrepo/unix-tutorial" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/pierrepo/unix-tutorial/issues/new?title=Issue%20on%20page%20%2Ftutorial_2/2_donnees.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../_sources/tutorial_2/2_donnees.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>


<script>
document.write(`
  <button class="theme-switch-button btn btn-sm btn-outline-primary navbar-btn rounded-circle" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch" data-mode="light"><i class="fa-solid fa-sun"></i></span>
    <span class="theme-switch" data-mode="dark"><i class="fa-solid fa-moon"></i></span>
    <span class="theme-switch" data-mode="auto"><i class="fa-solid fa-circle-half-stroke"></i></span>
  </button>
`);
</script>

<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Pr√©parer les donn√©es üóÉÔ∏è</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#creer-le-repertoire-de-travail">Cr√©er le r√©pertoire de travail</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#telecharger-les-donnees-de-sequencage">T√©l√©charger les donn√©es de s√©quen√ßage</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#methode-1-sra-run-selector">M√©thode 1 : SRA Run Selector</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#methode-2-sra-explorer">M√©thode 2 : SRA Explorer</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#telecharger-le-genome-de-reference-et-ses-annotations">T√©l√©charger le g√©nome de r√©f√©rence et ses annotations</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="preparer-les-donnees">
<h1>Pr√©parer les donn√©es üóÉÔ∏è<a class="headerlink" href="#preparer-les-donnees" title="Permalink to this heading">#</a></h1>
<p>L‚Äôarticle orginal publi√© en 2016 par <a class="reference external" href="https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1006453">Kelliher <em>et al.</em></a> indique dans la rubrique <em>Data Availability</em> :</p>
<blockquote>
<div><p>RNA-Sequencing gene expression data from this manuscript have been submitted to the NCBI Gene Expression Omnibus (GEO; <a class="reference external" href="http://www.ncbi.nlm.nih.gov/geo/">http://www.ncbi.nlm.nih.gov/geo/</a>) under accession number GSE80474.</p>
</div></blockquote>
<p>Le num√©ro du projet qui nous int√©resse est donc : <strong>GSE80474</strong></p>
<section id="creer-le-repertoire-de-travail">
<h2>Cr√©er le r√©pertoire de travail<a class="headerlink" href="#creer-le-repertoire-de-travail" title="Permalink to this heading">#</a></h2>
<p>Sur le cluster de l‚ÄôIFB, il ne faut pas travailler dans votre r√©pertoire personnel car l‚Äôespace disponible est tr√®s limit√©. Il faut travailler dans un r√©pertoire d√©di√© √† votre projet (ici, la formation DUO : <code class="docutils literal notranslate"><span class="pre">/shared/projects/202304_duo</span></code>)</p>
<p>Depuis l‚Äôinterface JupyterLab du cluster IFB, ouvrez un terminal, cr√©ez un r√©pertoire pour les donn√©es de ce tutoriel puis d√©placez-vous y :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>/shared/projects/202304_duo/<span class="nv">$USER</span>/rnaseq
<span class="nb">cd</span><span class="w"> </span>/shared/projects/202304_duo/<span class="nv">$USER</span>/rnaseq
</pre></div>
</div>
<p>Ici, <code class="docutils literal notranslate"><span class="pre">$USER</span></code> va automatiquement √™tre remplac√© par votre nom d‚Äôutilisateur. Vous n‚Äôavez rien √† faire.</p>
</section>
<section id="telecharger-les-donnees-de-sequencage">
<h2>T√©l√©charger les donn√©es de s√©quen√ßage<a class="headerlink" href="#telecharger-les-donnees-de-sequencage" title="Permalink to this heading">#</a></h2>
<p>L‚Äôarticle de Kelliher <em>et al.</em> nous a fourni le num√©ro du projet sur GEO. Cependant, l‚Äô√©tude a port√© sur deux organismes : <em>Saccharomyces cerevisiae</em> et <em>Cryptococcus neoformans var. grubii</em>. Nous ne sommes int√©ress√©s que par les donn√©es de <em>Saccharomyces cerevisiae</em> que nous allons devoir s√©lectionner.</p>
<section id="methode-1-sra-run-selector">
<h3>M√©thode 1 : SRA Run Selector<a class="headerlink" href="#methode-1-sra-run-selector" title="Permalink to this heading">#</a></h3>
<p>Dans l‚Äôoutil <a class="reference external" href="https://trace.ncbi.nlm.nih.gov/Traces/study/">SRA Run Selector</a>, entrez l‚Äôidentifiant du projet : GSE80474.</p>
<p>Un total de 74 <em>runs</em> sont disponibles. Cliquez alors sur le bouton gris <em>Metadata</em> correspondant au total.</p>
<p><strong>Sur votre machine du DU</strong></p>
<p>T√©l√©chargez le fichier <code class="docutils literal notranslate"><span class="pre">SraRunTable.txt</span></code> sur votre machine locale. Il s‚Äôagit d‚Äôun fichier CSV, c‚Äôest-√†-dire d‚Äôun fichier tabul√© avec des colonnes s√©par√©es par des virgules. Ouvrez-le avec Microsoft Excel ou LibreOffice Calc pour voir √† quoi il ressemble.</p>
<p><strong>Depuis l‚Äôinterface JupyterLab du cluster IFB</strong></p>
<p>Dans JupyterLab, utilisez l‚Äôexplorateur de fichiers (√† gauche) pour vous d√©placer dans le r√©pertoire que vous avez cr√©√© pr√©c√©demment (<code class="docutils literal notranslate"><span class="pre">/shared/projects/202304_duo/$USER/rnaseq</span></code>).</p>
<p>En cliquant sur l‚Äôic√¥ne <em>Upload Files</em>, importez le fichier <code class="docutils literal notranslate"><span class="pre">SraRunTable.txt</span></code> dans votre r√©pertoire projet.</p>
<p>S√©lectionnez les identifiants des <em>runs</em> correspondant √† <em>S. cerevisiae</em> avec la commande :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>grep<span class="w"> </span><span class="s2">&quot;Saccharomyces cerevisiae&quot;</span><span class="w"> </span>SraRunTable.txt<span class="w"> </span><span class="p">|</span><span class="w"> </span>cut<span class="w"> </span>-d<span class="s2">&quot;,&quot;</span><span class="w"> </span>-f1<span class="w"> </span>&gt;<span class="w"> </span>runs_scere.txt
</pre></div>
</div>
<p>Pouvez-vous expliquer ce que fait cette commande ?</p>
<div class="tip dropdown admonition">
<p class="admonition-title">Solution</p>
<p>La commande <code class="docutils literal notranslate"><span class="pre">grep</span> <span class="pre">&quot;Saccharomyces</span> <span class="pre">cerevisiae&quot;</span> <span class="pre">SraRunTable.txt</span></code> extraie du fichier <code class="docutils literal notranslate"><span class="pre">SraRunTable.txt</span></code> les lignes qui contiennent le texte <code class="docutils literal notranslate"><span class="pre">Saccharomyces</span> <span class="pre">cerevisiae</span></code>.</p>
<p>Le r√©sultat est ensuite envoy√© √† la commande <code class="docutils literal notranslate"><span class="pre">cut</span></code> via le pipe <code class="docutils literal notranslate"><span class="pre">|</span></code>.</p>
<p>La commande <code class="docutils literal notranslate"><span class="pre">cut</span> <span class="pre">-d&quot;,&quot;</span> <span class="pre">-f1</span></code> va extraire de chaque ligne le premier champ (<code class="docutils literal notranslate"><span class="pre">-f1</span></code>) s√©par√© par une virgule (<code class="docutils literal notranslate"><span class="pre">-d&quot;,&quot;</span></code>).</p>
<p>Le r√©sultat est enfin enregistr√© dans le fichier <code class="docutils literal notranslate"><span class="pre">runs_scere.txt</span></code> avec la rediction <code class="docutils literal notranslate"><span class="pre">&gt;</span></code>.</p>
</div>
<p>V√©rifiez que vous avez bien 50 <em>runs</em> de list√©s dans le fichier <code class="docutils literal notranslate"><span class="pre">runs_scere.txt</span></code> :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>wc<span class="w"> </span>-l<span class="w"> </span>runs_scere.txt<span class="w"> </span>
<span class="m">50</span><span class="w"> </span>runs_scere.txt
</pre></div>
</div>
<p>Cr√©ez ensuite un fichier qui ne va contenir que les 3 premiers √©chantillons avec la commande :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>grep<span class="w"> </span><span class="s2">&quot;Saccharomyces cerevisiae&quot;</span><span class="w"> </span>SraRunTable.txt<span class="w"> </span><span class="p">|</span><span class="w"> </span>cut<span class="w"> </span>-d<span class="s2">&quot;,&quot;</span><span class="w"> </span>-f1<span class="w"> </span><span class="p">|</span><span class="w"> </span>head<span class="w"> </span>-n<span class="w"> </span><span class="m">3</span><span class="w"> </span>&gt;<span class="w"> </span>runs_scere_small.txt
</pre></div>
</div>
<div class="admonition attention">
<p class="admonition-title">Attention</p>
<p>√Ä partir de maintenant, n‚Äôex√©cutez plus les commandes suivantes car le t√©l√©chargement et la compression des donn√©es vont prendre plusieurs dizaines de minutes. Lisez les commandes propos√©es et essayez de comprendre ce qu‚Äôelles font.</p>
<p>Reprenez l‚Äôex√©cution des commandes √† la m√©thode 2.</p>
</div>
<p>T√©l√©chargez les fichiers fastq associ√©s aux 3 premiers √©chantillons :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>reads
<span class="k">for</span><span class="w"> </span>sample<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="k">$(</span>cat<span class="w"> </span>runs_scere_small.txt<span class="k">)</span>
<span class="k">do</span><span class="w"> </span>
<span class="w">    </span><span class="nb">echo</span><span class="w"> </span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span>
<span class="w">    </span>fasterq-dump<span class="w"> </span>--progress<span class="w"> </span>--outdir<span class="w"> </span>reads<span class="w"> </span><span class="si">${</span><span class="nv">sample</span><span class="si">}</span>
<span class="k">done</span>
</pre></div>
</div>
<p>Cette commande va prendre plusieurs minutes, patientez.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>du<span class="w"> </span>-csh<span class="w"> </span>reads/*
<span class="m">4</span>,1G<span class="w">    </span>reads/SRR3405783.fastq
<span class="m">4</span>,7G<span class="w">    </span>reads/SRR3405784.fastq
<span class="m">4</span>,3G<span class="w">    </span>reads/SRR3405785.fastq
13G<span class="w">     </span>total
</pre></div>
</div>
<p>Pour √©conomiser un peu d‚Äôespace, compressez les fichiers fastq avec <code class="docutils literal notranslate"><span class="pre">gzip</span></code> :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>gzip<span class="w"> </span>reads/*
</pre></div>
</div>
<p>Cette commande va prendre quelques minutes, patientez.</p>
<p>V√©rifiez que vous avez gagn√© de l‚Äôespace :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>du<span class="w"> </span>-csh<span class="w"> </span>reads/*
864M<span class="w">    </span>reads/SRR3405783.fastq.gz
983M<span class="w">    </span>reads/SRR3405784.fastq.gz
910M<span class="w">    </span>reads/SRR3405785.fastq.gz
<span class="m">2</span>.7G<span class="w">    </span>total
</pre></div>
</div>
<p>On a gagn√© environ 10 Go d‚Äôespace disque, ce qui n‚Äôest pas n√©gligeable.</p>
</section>
<section id="methode-2-sra-explorer">
<h3>M√©thode 2 : SRA Explorer<a class="headerlink" href="#methode-2-sra-explorer" title="Permalink to this heading">#</a></h3>
<p>Le num√©ro du projet GSE80474 commence par les lettres <code class="docutils literal notranslate"><span class="pre">GSE</span></code> ce qui nous indique que c‚Äôest un projet initialement d√©pos√© dans la base de donn√©es GEO. Cette base n‚Äô√©tant pas toujours bien prise en charge par l‚Äôoutil sra-explorer, nous allons tout d‚Äôabord r√©cup√©rer sur le site SRA Run Selector l‚Äôidentifiant <em>BioProject</em> correspondant.</p>
<p>Sur le site <a class="reference external" href="https://trace.ncbi.nlm.nih.gov/Traces/study/">SRA Run Selector</a> :</p>
<ol class="arabic simple">
<li><p>Entrez l‚Äôidentifiant du projet : GSE80474.</p></li>
<li><p>Cliquez sur le bouton bleu <em>Search</em>.</p></li>
<li><p>Dans la rubrique <em>Common Fields</em> (tout en haut), r√©cup√©rez l‚Äôidentifiant <em>BioProject</em> : PRJNA319029</p></li>
</ol>
<p>C‚Äôest avec cet identifiant BioProject que nous allons r√©cup√©rer les donn√©es.</p>
<p>Sur le site <a class="reference external" href="https://sra-explorer.info/">SRA EXplorer</a> :</p>
<ol class="arabic simple">
<li><p>Indiquez le num√©ro du <em>BioProject</em>, ici PRJNA319029, puis cliquez sur le petite loupe pour lancer la recherche.</p></li>
<li><p>Vous obtenez ensuite 74 r√©ponses qui correspondent aux diff√©rents fichiers / √©chantillons.</p></li>
<li><p>Affinez les r√©ponses en tapant ¬´ Scerevisiae ¬ª dans le champ ¬´ Filter results: ¬ª. Vous devriez obtenir 50 r√©sultats.</p></li>
<li><p>S√©lectionnez tous les r√©sultats en cliquant sur le case vide √† droite de <em>Title</em>.</p></li>
<li><p>Cliquez sur le bouton ¬´ Add 50 to collection ¬ª.</p></li>
<li><p>Cliquez ensuite en haut √† droite sur le bouton bleu ¬´ 50 saved datasets ¬ª.</p></li>
<li><p>Cliquez enfin sur ¬´ Bash script for downloading FastQ files ¬ª.</p></li>
<li><p>T√©l√©chargez sur votre <strong>machine locale</strong> le script qui vous permettra de t√©l√©charger tous les fichiers fastq (<code class="docutils literal notranslate"><span class="pre">sra_explorer_fastq_download.sh</span></code>).</p></li>
<li><p>Importer ce script dans l‚Äôinterface JupyterLab du cluster IFB, dans votre r√©pertoire projet.</p></li>
</ol>
<p>Voici les 5 premi√®res lignes du script t√©l√©charg√© :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>head<span class="w"> </span>-n<span class="w"> </span><span class="m">5</span><span class="w"> </span>sra_explorer_fastq_download.sh
<span class="c1">#!/usr/bin/env bash</span>
curl<span class="w"> </span>-L<span class="w"> </span>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR340/009/SRR3405789/SRR3405789.fastq.gz<span class="w"> </span>-o<span class="w"> </span>SRR3405789_GSM2128026_Scerevisiae_YEPD_aF_30min_Saccharomyces_cerevisiae_RNA-Seq.fastq.gz
curl<span class="w"> </span>-L<span class="w"> </span>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR340/001/SRR3405791/SRR3405791.fastq.gz<span class="w"> </span>-o<span class="w"> </span>SRR3405791_GSM2128028_Scerevisiae_YEPD_aF_40min_Saccharomyces_cerevisiae_RNA-Seq.fastq.gz
curl<span class="w"> </span>-L<span class="w"> </span>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR340/004/SRR3405784/SRR3405784.fastq.gz<span class="w"> </span>-o<span class="w"> </span>SRR3405784_GSM2128021_Scerevisiae_YEPD_aF_5min_Saccharomyces_cerevisiae_RNA-Seq.fastq.gz
curl<span class="w"> </span>-L<span class="w"> </span>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR340/003/SRR3405783/SRR3405783.fastq.gz<span class="w"> </span>-o<span class="w"> </span>SRR3405783_GSM2128020_Scerevisiae_YEPD_aF_0min_Saccharomyces_cerevisiae_RNA-Seq.fastq.gz
</pre></div>
</div>
<p>C‚Äôest bien un script Bash car la premi√®re ligne est <code class="docutils literal notranslate"><span class="pre">#!/usr/bin/env</span> <span class="pre">bash</span></code>. Ensuite, chaque ligne qui d√©bute par <code class="docutils literal notranslate"><span class="pre">curl</span></code> t√©l√©charge un fichier .fastq.gz. La syntaxe de la commande <code class="docutils literal notranslate"><span class="pre">curl</span></code> est la suivante :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>curl<span class="w"> </span>-L<span class="w"> </span>ADRESSE-DU-FICHIER-√Ä-T√âL√âCHARGER<span class="w"> </span>-o<span class="w"> </span>NOM-DU-FICHIER-SUR-LE-DISQUE-LOCAL
</pre></div>
</div>
<p>Nous aimerions modifier ce script pour faire en sorte que</p>
<ol class="arabic simple">
<li><p>Le nom du fichier enregistr√© localement ne contienne que le num√©ro d‚Äôaccession du fichier, tel que pr√©sent sur les serveurs de SRA (par exemple : <code class="docutils literal notranslate"><span class="pre">SRR3405789</span></code>) et pas les m√©tadonnn√©es associ√©es (par exemple : <code class="docutils literal notranslate"><span class="pre">_GSM2128026_Scerevisiae_YEPD_aF_30min_Saccharomyces_cerevisiae_RNA-Seq.fastq.gz</span></code>). Pour cela, il faut remplacer l‚Äôoption <code class="docutils literal notranslate"><span class="pre">-o</span></code> par <code class="docutils literal notranslate"><span class="pre">-O</span></code> (sans argument).</p></li>
<li><p>Tous les fichiers soient enregistr√©s dans le m√™me r√©pertoire (par exemple <code class="docutils literal notranslate"><span class="pre">reads</span></code>). Il faut alors ajouter l‚Äôoption <code class="docutils literal notranslate"><span class="pre">--output-dir</span></code> avec l‚Äôargument <code class="docutils literal notranslate"><span class="pre">reads</span></code>.</p></li>
</ol>
<p>Nous utilisons ici la commande <code class="docutils literal notranslate"><span class="pre">sed</span></code> qui peut modifier √† la vol√©e les lignes d‚Äôun fichier :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>sed<span class="w"> </span>-E<span class="w"> </span><span class="s1">&#39;s/-o .*/-O --output-dir reads/&#39;</span><span class="w"> </span>sra_explorer_fastq_download.sh<span class="w">  </span>&gt;<span class="w"> </span>sra_explorer_fastq_download_2.sh
</pre></div>
</div>
<p>Voici les 5 premi√®res lignes du script <code class="docutils literal notranslate"><span class="pre">sra_explorer_fastq_download_2.sh</span></code> :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>head<span class="w"> </span>-n<span class="w"> </span><span class="m">5</span><span class="w"> </span>sra_explorer_fastq_download_2.sh
<span class="c1">#!/usr/bin/env bash</span>
curl<span class="w"> </span>-L<span class="w"> </span>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR340/009/SRR3405789/SRR3405789.fastq.gz<span class="w"> </span>-O<span class="w"> </span>--output-dir<span class="w"> </span>reads
curl<span class="w"> </span>-L<span class="w"> </span>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR340/001/SRR3405791/SRR3405791.fastq.gz<span class="w"> </span>-O<span class="w"> </span>--output-dir<span class="w"> </span>reads
curl<span class="w"> </span>-L<span class="w"> </span>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR340/004/SRR3405784/SRR3405784.fastq.gz<span class="w"> </span>-O<span class="w"> </span>--output-dir<span class="w"> </span>reads
curl<span class="w"> </span>-L<span class="w"> </span>ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR340/003/SRR3405783/SRR3405783.fastq.gz<span class="w"> </span>-O<span class="w"> </span>--output-dir<span class="w"> </span>reads
</pre></div>
</div>
<p>Pour v√©rifier que notre script fonctionne, nous allons t√©l√©charger les 3 premiers fichiers <em>.fastq.gz</em>. Pour cela, cr√©ez un script interm√©diaire en ne s√©lectionnant que les 4 premi√®res lignes du script de t√©l√©chargement :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>head<span class="w"> </span>-n<span class="w"> </span><span class="m">4</span><span class="w"> </span>sra_explorer_fastq_download_2.sh<span class="w"> </span>&gt;<span class="w"> </span>sra_explorer_fastq_download_2_small.sh
</pre></div>
</div>
<p>Le script fonctionne avec un version r√©cente de <code class="docutils literal notranslate"><span class="pre">curl</span></code>, chargez cette version avec :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>module<span class="w"> </span>load<span class="w"> </span>curl
</pre></div>
</div>
<p>V√©rifiez que c‚Äôest bien le cas :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>curl<span class="w"> </span>--version<span class="w"> </span><span class="p">|</span><span class="w"> </span>head<span class="w"> </span>-n<span class="w"> </span><span class="m">1</span>
curl<span class="w"> </span><span class="m">7</span>.80.0<span class="w"> </span><span class="o">(</span>x86_64-conda-linux-gnu<span class="o">)</span><span class="w"> </span>libcurl/7.80.0<span class="w"> </span>OpenSSL/3.0.0<span class="w"> </span>zlib/1.2.11<span class="w"> </span>libssh2/1.10.0<span class="w"> </span>nghttp2/1.43.0
</pre></div>
</div>
<p>Enfin, lancez le script pour t√©l√©charger les donn√©es :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>reads
bash<span class="w"> </span>sra_explorer_fastq_download_2_small.sh
</pre></div>
</div>
<p>Patientez quelques minutes que le t√©l√©chargement se termine.</p>
<div class="admonition hint">
<p class="admonition-title">Hint</p>
<p>text
Le script t√©l√©charge directement les donn√©es compress√©es, ce qui est beaucoup plus rapide que de t√©l√©charger les donn√©es non compress√©es puis de les compresser.</p>
</div>
<p>Calculez l‚Äôespace occup√© par les donn√©es :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>du<span class="w"> </span>-csh<span class="w"> </span>reads/*
776M<span class="w">    </span>reads/SRR3405783.fastq.gz
883M<span class="w">    </span>reads/SRR3405784.fastq.gz
818M<span class="w">    </span>reads/SRR3405785.fastq.gz
<span class="m">2</span>.5G<span class="w">    </span>total
</pre></div>
</div>
</section>
</section>
<section id="telecharger-le-genome-de-reference-et-ses-annotations">
<h2>T√©l√©charger le g√©nome de r√©f√©rence et ses annotations<a class="headerlink" href="#telecharger-le-genome-de-reference-et-ses-annotations" title="Permalink to this heading">#</a></h2>
<p>On trouve dans le fichier <em>S1 Supporting Information Methods</em> la desciption du g√©nome de <em>S. cerevisiae</em> utilis√© :</p>
<blockquote>
<div><p>The S. cerevisiae S288C genome (Ensembl build R64-1-1) was downloaded from Illumina iGenomes on March 2, 2016 (<a class="reference external" href="https://support.illumina.com/sequencing/sequencing_software/igenome.html">https://support.illumina.com/sequencing/sequencing_software/igenome.html</a>).</p>
</div></blockquote>
<p>T√©l√©chargez le fichier concern√© (<code class="docutils literal notranslate"><span class="pre">Ensembl</span> <span class="pre">build</span> <span class="pre">R64-1-1.tar.gz</span></code>) et d√©compressez l‚Äôarchive :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>wget<span class="w"> </span>http://igenomes.illumina.com.s3-website-us-east-1.amazonaws.com/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Saccharomyces_cerevisiae_Ensembl_R64-1-1.tar.gz
tar<span class="w"> </span>zxvf<span class="w"> </span>Saccharomyces_cerevisiae_Ensembl_R64-1-1.tar.gz
</pre></div>
</div>
<p>R√©cup√©rez ensuite les fichiers contenant le g√©nome et les annotations :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mkdir<span class="w"> </span>-p<span class="w"> </span>genome
cp<span class="w"> </span>Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf<span class="w"> </span>genome
cp<span class="w"> </span>Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/WholeGenomeFasta/genome.fa<span class="w"> </span>genome
</pre></div>
</div>
<p>Supprimez enfin le r√©pertoire <code class="docutils literal notranslate"><span class="pre">Saccharomyces_cerevisiae</span></code> et l‚Äôarchive contenant le g√©nome qui ne nous int√©ressent plus :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>rm<span class="w"> </span>-rf<span class="w"> </span>Saccharomyces_cerevisiae<span class="w"> </span>Saccharomyces_cerevisiae_Ensembl_R64-1-1.tar.gz<span class="w"> </span>README.txt
</pre></div>
</div>
<p>In fine, vous devriez obtenir l‚Äôorganisation de fichiers suivante :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$<span class="w"> </span>tree
.
‚îú‚îÄ‚îÄ<span class="w"> </span>genome
‚îÇ<span class="w">   </span>‚îú‚îÄ‚îÄ<span class="w"> </span>genes.gtf
‚îÇ<span class="w">   </span>‚îî‚îÄ‚îÄ<span class="w"> </span>genome.fa
‚îú‚îÄ‚îÄ<span class="w"> </span>reads
‚îÇ<span class="w">   </span>‚îú‚îÄ‚îÄ<span class="w"> </span>SRR3405783.fastq.gz
‚îÇ<span class="w">   </span>‚îú‚îÄ‚îÄ<span class="w"> </span>SRR3405784.fastq.gz
‚îÇ<span class="w">   </span>‚îî‚îÄ‚îÄ<span class="w"> </span>SRR3405785.fastq.gz
‚îú‚îÄ‚îÄ<span class="w"> </span>runs_scere_small.txt
‚îú‚îÄ‚îÄ<span class="w"> </span>runs_scere.txt
‚îú‚îÄ‚îÄ<span class="w"> </span>sra_explorer_fastq_download_2.sh
‚îú‚îÄ‚îÄ<span class="w"> </span>sra_explorer_fastq_download_2_small.sh
‚îú‚îÄ‚îÄ<span class="w"> </span>sra_explorer_fastq_download.sh
‚îî‚îÄ‚îÄ<span class="w"> </span>SraRunTable.txt

<span class="m">2</span><span class="w"> </span>directories,<span class="w"> </span><span class="m">11</span><span class="w"> </span>files
</pre></div>
</div>
<p>Les options <code class="docutils literal notranslate"><span class="pre">--du</span> <span class="pre">-h</span></code> de <code class="docutils literal notranslate"><span class="pre">tree</span></code> sont pratiques pour afficher aussi la taille des fichiers et des r√©pertoires :</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span><span class="w"> </span>tree<span class="w"> </span>--du<span class="w"> </span>-h
.
‚îú‚îÄ‚îÄ<span class="w"> </span><span class="o">[</span><span class="w"> </span>23M<span class="o">]</span><span class="w">  </span>genome
‚îÇ<span class="w">   </span>‚îú‚îÄ‚îÄ<span class="w"> </span><span class="o">[</span><span class="w"> </span>11M<span class="o">]</span><span class="w">  </span>genes.gtf
‚îÇ<span class="w">   </span>‚îî‚îÄ‚îÄ<span class="w"> </span><span class="o">[</span><span class="w"> </span>12M<span class="o">]</span><span class="w">  </span>genome.fa
‚îú‚îÄ‚îÄ<span class="w"> </span><span class="o">[</span><span class="m">2</span>.4G<span class="o">]</span><span class="w">  </span>reads
‚îÇ<span class="w">   </span>‚îú‚îÄ‚îÄ<span class="w"> </span><span class="o">[</span>776M<span class="o">]</span><span class="w">  </span>SRR3405783.fastq.gz
‚îÇ<span class="w">   </span>‚îú‚îÄ‚îÄ<span class="w"> </span><span class="o">[</span>883M<span class="o">]</span><span class="w">  </span>SRR3405784.fastq.gz
‚îÇ<span class="w">   </span>‚îî‚îÄ‚îÄ<span class="w"> </span><span class="o">[</span>817M<span class="o">]</span><span class="w">  </span>SRR3405785.fastq.gz
‚îú‚îÄ‚îÄ<span class="w"> </span><span class="o">[</span><span class="w">  </span><span class="m">32</span><span class="o">]</span><span class="w">  </span>runs_scere_small.txt
‚îú‚îÄ‚îÄ<span class="w"> </span><span class="o">[</span><span class="w"> </span><span class="m">550</span><span class="o">]</span><span class="w">  </span>runs_scere.txt
‚îú‚îÄ‚îÄ<span class="w"> </span><span class="o">[</span><span class="m">5</span>.2K<span class="o">]</span><span class="w">  </span>sra_explorer_fastq_download_2.sh
‚îú‚îÄ‚îÄ<span class="w"> </span><span class="o">[</span><span class="w"> </span><span class="m">341</span><span class="o">]</span><span class="w">  </span>sra_explorer_fastq_download_2_small.sh
‚îú‚îÄ‚îÄ<span class="w"> </span><span class="o">[</span><span class="m">8</span>.7K<span class="o">]</span><span class="w">  </span>sra_explorer_fastq_download.sh
‚îî‚îÄ‚îÄ<span class="w"> </span><span class="o">[</span><span class="w"> </span>29K<span class="o">]</span><span class="w">  </span>SraRunTable.txt

<span class="w"> </span><span class="m">2</span>.4G<span class="w"> </span>used<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">2</span><span class="w"> </span>directories,<span class="w"> </span><span class="m">11</span><span class="w"> </span>files
</pre></div>
</div>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./tutorial_2"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
                <footer class="bd-footer-article">
                  
<div class="footer-article-items footer-article__inner">
  
    <div class="footer-article-item"><!-- Previous / next buttons -->
<div class="prev-next-area">
</div></div>
  
</div>

                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#creer-le-repertoire-de-travail">Cr√©er le r√©pertoire de travail</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#telecharger-les-donnees-de-sequencage">T√©l√©charger les donn√©es de s√©quen√ßage</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#methode-1-sra-run-selector">M√©thode 1 : SRA Run Selector</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#methode-2-sra-explorer">M√©thode 2 : SRA Explorer</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#telecharger-le-genome-de-reference-et-ses-annotations">T√©l√©charger le g√©nome de r√©f√©rence et ses annotations</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Pierre Poulain
</p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
<div class="extra_footer">
  <p>
Ce contenu est mis √† disposition selon les termes de la licence <a href="https://creativecommons.org/licenses/by-sa/4.0/deed.fr" target="_blank">Creative Commons Attribution - Partage dans les M√™mes Conditions 4.0 International</a> (CC BY-SA 4.0).
</p>

</div>
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=e353d410970836974a52"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=e353d410970836974a52"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>